selfdrive/ui/ui.cc

```diff
@@ -339,8 +337,9 @@ void Device::updateBrightness(const UIState &s) {
   }
 
   if (brightness != last_brightness) {
-    if (!brightness_future.isRunning()) {
-      brightness_future = QtConcurrent::run(Hardware::set_brightness, brightness);
+    if (!brightness_future.valid() ||
```

Comment:
how about we just make this a normal thread? I think that's even simpler than a `std::future`

Reply:
Overall, std::async is simpler to use compared to std::thread for handling this type of asynchronous process because it manages threads and synchronization automatically, making the code easier to write and maintain.  It also makes exception handling easier, as any exceptions thrown by set_brightness are captured and can be handled through the future.


Prototypical Code Using std::thread:

cass Device {
   std::unique_ptr<std::thread> brightness_thread;
}

void Device::updateBrightness(const UIState &s) {
  f (brightness != last_brightness) {
        if (!brightness_thread || !brightness_thread->joinable())
            brightness_thread = std::make_unique<std::thread>(Hardware::set_brightness, brightness);
            last_brightness = brightness; 
        }
}

Device::~Device() {
if (brightness_thread && brightness_thread->joinable()){
 brightness_thread->join();
}
}

Reply:
we don't use cpp exceptions though. I was thinking a long-running thread that sticks around for the life of ui